### PR TITLE
Create Profile Page

### PR DESCRIPTION
**Pull request by jpark799**
http://localhost:3000/settings/summary

a little ugly but the functionality is there for at least a demo

image

### CODE REVIEW

File: packages/web-app/src/modules/profile/ProfileStore.ts [hunk 1]
```diff
@@ -37,6 +48,41 @@ export class ProfileStore {
...
+    console.log('Loading avatars')
+    this.isLoading = true
+    try {
+      let avatar = yield this.axios.get('/api/v2/avatars')
+      this.avatars = avatar.data
+      this.formAvatars = avatar.data.slice()
+    } catch (err) {
+      console.error('Avatars error: ', err)
+      this.avatars = undefined
+      // this.store.routing.replace('/avatars-error')
```
Comment (seniorquico) [inline packages/web-app/src/modules/profile/ProfileStore.ts:62]: Remove commented out code.

File: packages/web-app/src/modules/profile/ProfileStore.ts [hunk 2]
```diff
@@ -37,6 +48,41 @@ export class ProfileStore {
...
+    console.log('Loading selected avatar')
+
+    this.isLoading = true
+    try {
+      let selectedAvatar = yield this.axios.get('/api/v2/avatars/selected')
+      console.log(selectedAvatar, 'SELECTED AVATAR')
+      this.selectedAvatar = selectedAvatar.data
+    } catch (err) {
+      console.error('selectedAvatars error: ', err)
+      this.selectedAvatar = undefined // TODO: change this to an defaultAvatar object
```
Comment (seniorquico) [inline packages/web-app/src/modules/profile/ProfileStore.ts:79]: Fix the TODO task.

File: packages/web-app/src/modules/profile/ProfileStore.ts [hunk 3]
```diff
@@ -37,6 +48,41 @@ export class ProfileStore {
...
+  })
+
+  @action.bound
+  loadSelectedAvatar = flow(function* (this: ProfileStore) {
+    console.log('Loading selected avatar')
+
+    this.isLoading = true
+    try {
+      let selectedAvatar = yield this.axios.get('/api/v2/avatars/selected')
+      console.log(selectedAvatar, 'SELECTED AVATAR')
```
Comment (seniorquico) [inline packages/web-app/src/modules/profile/ProfileStore.ts:75]: Let's drop debug logging. It's not generally helpful. We'll already get errors logged by axios responses.

File: packages/web-app/src/modules/profile/ProfileStore.ts [hunk 4]
```diff
@@ -37,6 +48,41 @@ export class ProfileStore {
...
+      this.selectedAvatar = selectedAvatar.data
+    } catch (err) {
+      console.error('selectedAvatars error: ', err)
```
Comment (seniorquico) [inline packages/web-app/src/modules/profile/ProfileStore.ts:78]: Drop debug logging.

File: packages/web-app/src/modules/profile/ProfileStore.ts [hunk 5]
```diff
@@ -37,6 +48,41 @@ export class ProfileStore {
...
+      this.avatars = undefined
+      // this.store.routing.replace('/avatars-error')
+    } finally {
+      this.isLoading = false
+    }
+  })
+
+  @action.bound
+  loadSelectedAvatar = flow(function* (this: ProfileStore) {
+    console.log('Loading selected avatar')
```
Comment (seniorquico) [inline packages/web-app/src/modules/profile/ProfileStore.ts:70]: Drop debug logging.

File: packages/web-app/src/modules/profile/ProfileStore.ts [hunk 6]
```diff
@@ -62,13 +108,42 @@ export class ProfileStore {
...
+  selectAvatar = flow(function* (this: ProfileStore, id: string) {
+    this.isUpdating = true
+
+    try {
+      let patch = yield this.axios.patch('/api/v2/avatars/selected', { avatarId: id })
+      let selectedAvatar = patch.data
+
+      this.selectedAvatar = selectedAvatar
+    } catch (err) {
+      console.error('select avatar error: ', err)
```
Comment (seniorquico) [inline packages/web-app/src/modules/profile/ProfileStore.ts:120]: Drop debug logging.

File: packages/web-app/src/modules/account-views/account-views/AccountContainer.ts [hunk 7]
```diff
@@ -4,9 +4,13 @@ import { Account } from './components'
 
 const mapStoreToProps = (store: RootStore): any => ({
   profile: store.profile.currentProfile,
+  avatars: store.profile.avatars,
+  selectedAvatar: store.profile.selectedAvatar,
   isUpdating: store.profile.isUpdating,
   onUpdateUsername: store.profile.updateUsername,
   onUpdateMinecraftUsername: store.profile.updateMinecraftUsername,
+  onSelectAvatar: store.profile.selectAvatar,
+  clearAvatarError: store.profile.clearAvatarError,
```
Comment (seniorquico) [inline packages/web-app/src/modules/account-views/account-views/AccountContainer.ts:13]:
```suggestion
  onClearAvatarError: store.profile.clearAvatarError,
```

**Review by seniorquico — CHANGES_REQUESTED** [review]

File: packages/web-app/src/modules/profile/ProfileStore.ts [hunk 8]
```diff
@@ -37,6 +49,43 @@ export class ProfileStore {
     return this.currentProfile
   })
 
+  @action.bound
+  loadAvatars = flow(function* (this: ProfileStore) {
+    console.log('Loading avatars')
```
Comment (jpark799) [inline packages/web-app/src/modules/profile/ProfileStore.ts:53]: @seniorquico can i remove these logs as well?

**Review by seniorquico — APPROVED** [review]

**Pull request merged**